src/client/csfle/state_machine.rs

```diff
@@ -181,6 +209,71 @@ impl CryptExecutor {
     }
 }
 
+#[derive(Debug)]
+struct Mongocryptd {
+    opts: MongocryptdOptions,
+    client: Client,
+    child: Mutex<Result<Process>>,
+}
+
+impl Mongocryptd {
+    async fn new(opts: MongocryptdOptions) -> Result<Self> {
+        let child = Mutex::new(Ok(Self::spawn(&opts)?));
+        let uri = opts.uri.as_deref().unwrap_or("mongodb://localhost:27020");
+        let mut options = crate::options::ClientOptions::parse_uri(uri, None).await?;
+        options.server_selection_timeout = Some(std::time::Duration::from_millis(10_000));
+        let client = Client::with_options(options)?;
+        Ok(Self {
+            opts,
+            client,
+            child,
+        })
+    }
+
+    async fn respawn(&self) -> Result<()> {
+        let mut child = match self.child.try_lock() {
+            Ok(l) => l,
+            _ => {
+                // Another respawn is in progress.  Lock to wait for it.
+                return unit_err(&*self.child.lock().await);
+            }
+        };
+        *child = Self::spawn(&self.opts);
```

Comment:
I think we may want to do an explicit `wait` here on the prior child process (if the result is `Ok`) to ensure its resources were reaped properly before starting a new one. I think the `try_wait` in the `Drop` of process isn't necessarily sufficient to ensure reaping happened.

Reply:
Done.